generator/generator.py

```diff
@@ -238,18 +238,16 @@ def generate_boards_json(input_data, arduino_cli_path):
     for fqbn, data in input_data.items():
         simple_fqbn = fqbn.replace(":", ".")
 
-        binary_path = f"../firmwares/loader/{simple_fqbn}/"
-        binary = (
-            Path(__file__).parent / binary_path / os.listdir(binary_path)[0]
+        loader_dir = f"../firmwares/loader/{simple_fqbn}/"
+        loader_path = (
+            Path(__file__).parent / loader_dir / os.listdir(loader_dir)[0]
```

Comment:
Am not sure I understand this line, why are you taking the first result of `os.listdir`?

I don't think this a good thing to do, the documentation says this:
> The list is in arbitrary order, and does not include the special entries '.' and '..' even if they are present in the directory.

https://docs.python.org/3/library/os.html#os.listdir

Also might as well use `Path.iterdir()` since we're using `pathlib`: https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir